crates/wasmparser/src/operators_validator.rs

```diff
@@ -578,10 +580,23 @@ impl OperatorValidator {
             }
             Operator::Else => {
                 let frame = self.pop_ctrl(resources)?;
-                if frame.kind != FrameKind::If {
-                    bail_op_err!("else found outside of an `if` block");
+                // The `catch_all` instruction shares an opcode with `else`,
+                // so we check the frame to see how it's interpreted.
+                if self.features.exceptions
+                    && (frame.kind == FrameKind::Try || frame.kind == FrameKind::Catch)
```

Comment:
Perhaps this could be:

```rust
match frame.kind {
    FrameKind::If => ...,
    FrameKind::Try | FrameKind::Catch => ..., // can assume `self.features.exceptions` is true
    _ => bail_op_err!(...),
}
```

To make it a bit more tidy?

I think we can skip the recheck of `self.features.exceptions` because that should be checked pushing the frame on the initial instruction.

Reply:
Thanks for the suggestion!